modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/portlet/action/GetLayoutReportsIssuesMVCResourceCommand.java

```diff
@@ -0,0 +1,167 @@
+import com.liferay.layout.reports.web.internal.constants.LayoutReportsPortletKeys;
+import com.liferay.layout.reports.web.internal.data.provider.LayoutReportsDataProvider;
+import com.liferay.layout.reports.web.internal.model.LayoutReportsIssue;
+import com.liferay.layout.seo.kernel.LayoutSEOLinkManager;
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.util.Http;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Cristina González
+ */
+@Component(
+	configurationPid = "com.liferay.layout.reports.web.internal.configuration.LayoutReportsPageSpeedConfiguration",
+	immediate = true,
+	property = {
+		"javax.portlet.name=" + LayoutReportsPortletKeys.LAYOUT_REPORTS,
+		"mvc.command.name=/layout_reports/get_layout_reports_issues"
+	},
+	service = MVCResourceCommand.class
+)
+public class GetLayoutReportsIssuesMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		_layoutReportsGooglePageSpeedConfigurationProvider =
+			new LayoutReportsGooglePageSpeedConfigurationProvider(
+				_configurationProvider,
+				ConfigurableUtil.createConfigurable(
+					LayoutReportsGooglePageSpeedConfiguration.class,
+					properties));
+	}
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		Locale locale = _portal.getLocale(resourceRequest);
+
+		try {
+			long groupId = ParamUtil.getLong(resourceRequest, "groupId");
+
+			Group group = _groupLocalService.fetchGroup(groupId);
+
+			if (group == null) {
+				_log.error("No API Key configured for the site " + groupId);
```

Comment:
I think this message is misleading. The problem is that the group does not exist. 

I tried calling the command without setting an API Key and it works fine (I guess with the restriction of no more than 1 or 2 requests per second).

Reply:
Hey @4lejandrito !!

Yeah, we can call the service without APIKey, but we have decided that this is NOT OK for the client, and the client will need a configured APIKey in order to call the PageSpeed API:

**Given** the Page Audit Panel
**When** PageSpeed's API Key is not configured and the user has no permissions for accessing to the site/system/instance settings
**Then** a message and an image show: Check issues that impact on your page's accessibility and SEO. Connect with PageSpeed from Instance or System Site settings > Pages > Google PageSpeed.

But you are totally right about the message, I will change it :=

Thanks!!!